cmd/kube-controllers/main.go

```diff
@@ -47,39 +47,45 @@ import (
 
 // VERSION is filled out during the build process (using git describe output)
 var VERSION string
+var version bool
 
-func main() {
-	// Configure log formatting.
-	log.SetFormatter(&logutils.Formatter{})
-
-	// Install a hook that adds file/line no information.
-	log.AddHook(&logutils.ContextHook{})
-
+func init() {
 	// Tell glog (used by client-go) to log into STDERR. Otherwise, we risk
 	// certain kinds of API errors getting logged into a directory not
 	// available in a `FROM scratch` Docker container, causing glog to abort
-	// hard with an exit code > 0.
 	err := flag.Set("logtostderr", "true")
 	if err != nil {
-		log.WithError(err).Fatal("Failed configure logging")
+		log.WithError(err).Fatal("Failed to set logging configuration")
 	}
 
-	// If `-v` is passed, display the version and exit.
-	// Use a new flag set so as not to conflict with existing libraries which use "flag"
-	flagSet := pflag.NewFlagSet("Calico", pflag.ExitOnError)
-	version := flagSet.BoolP("version", "v", false, "Display version")
-	err = flagSet.Parse(os.Args[1:])
+	// Add a flag to check the version.
+	flag.BoolVar(&version, "version", false, "Display version")
+
+	// Also tell klog to log to STDERR.
+	var flags flag.FlagSet
+	klog.InitFlags(&flags)
```

Comment:
Hey @caseydavenport what was the issue that that `klog` fixes?

Reply:
klog is just another log library used by some of our imports, and without setting its flags it logs to file by default...